Add some notes about potential issues with diagnostic tools (#21983)
dotnet/docs: docs/core/diagnostics/dotnet-gcdump.md

Collects a GC dump from a currently running process.

> [!WARNING]

> > This will trigger a gen-2 (full) garbage collection to the application to walk the GC heap which can suspend the runtime for a long time, especially when the GC heap is large. Do not use this command in performance-sensitive environments in such cases.

Reviewer: ```suggestion
> To walk the GC heap, this command triggers a generation 2 (full) garbage collection, which can suspend the runtime for a long time, especially when the GC heap is large. Don't use this command in performance-sensitive environments when the GC heap is large.
```
